### ISSUE TITLE
Keras installation get error that the system was unable to find the specified registry key or value 

### ISSUE THREAD

**Issue by idohi** [keras-team/keras#13209]
OS: Windows10
python: 3.6.8
conda: 4.5.4

To reproduce:
1. open anaconda prompt
2. run: `conda activate <my_environment>`
3. run : `conda install -c conda-forge keras`

Output:
```
conda install -c conda-forge keras
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 4.5.4
  latest version: 4.7.11

Please update conda by running

    $ conda update -n base conda



## Package Plan ##

  environment location: C:\Users\Ido\AppData\Local\conda\conda\envs\py36

  added / updated specs:
    - keras


The following NEW packages will be INSTALLED:

    keras:         2.2.4-py36_1            conda-forge
    libgpuarray:   0.7.6-hfa6e2cd_1003     conda-forge
    mako:          1.1.0-py_0              conda-forge
    pygpu:         0.7.6-py36h452e1ab_1000 conda-forge
    theano:        1.0.4-py36h6538335_1000 conda-forge
    vs2015_win-64: 14.0.25420-h55c1224_11

Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done

(py36) C:\Users\Ido>set "KERAS_BACKEND="

(py36) C:\Users\Ido>python C:\Users\Ido\AppData\Local\conda\conda\envs\py36\etc\keras\load_config.py  1>temp.txt

(py36) C:\Users\Ido>set /p KERAS_BACKEND= 0<temp.txt

(py36) C:\Users\Ido>del temp.txt

(py36) C:\Users\Ido>python -c "import keras"  1>nul 2>&1

(py36) C:\Users\Ido>if errorlevel 1 (
ver  1>nul
 set "KERAS_BACKEND=theano"
 python -c "import keras"  1>nul 2>&1
)

(py36) C:\Users\Ido>SET DISTUTILS_USE_SDK=1

(py36) C:\Users\Ido>SET MSSdk=1

(py36) C:\Users\Ido>SET platform=

(py36) C:\Users\Ido>IF /I [AMD64] == [amd64] set "platform=true"

(py36) C:\Users\Ido>IF /I [] == [amd64] set "platform=true"

(py36) C:\Users\Ido>if defined platform (set "VSREGKEY=HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\Microsoft\VisualStudio\14.0" )  ELSE (set "VSREGKEY=HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\VisualStudio\14.0" )

(py36) C:\Users\Ido>for /F "skip=2 tokens=2,*" %A in ('reg query "HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\Microsoft\VisualStudio\14.0" /v InstallDir') do SET "VSINSTALLDIR=%B"
ERROR: The system was unable to find the specified registry key or value.

(py36) C:\Users\Ido>if "" == "" (set "VSINSTALLDIR=" )

(py36) C:\Users\Ido>if "" == "" (
ECHO "WARNING: Did not find VS in registry or in VS140COMNTOOLS env var - your compiler may not work"
 GOTO End
)
"WARNING: Did not find VS in registry or in VS140COMNTOOLS env var - your compiler may not work"
The system cannot find the batch label specified - End
```

**Comment by minhdvo** [comment]
I met a similar problem even though I still can use python and other packages. 
I guess it could be related to administration permission. Because this problem occurred after installation and beginning the verification process for installing keras. Any suggestion?

**Comment by dmariko-yseop** [comment]
It seems to be calling the compiler from VS 2015 (version 14), exclusively ?

**Comment by RobeeF** [comment]
Same error for me. I still can use Python but I am not able to launch Spyder anymore

**Comment by sidney-cardozo** [comment]
I am having the same issue, has anyone else found a work around? @dmariko-yseop were able you selectively install VS 2015 version 14 and then get it to install?

**Comment by g3n3ralb3n-wp** [comment]
This even happens when you try and just install theano without keras.

**Comment by nocluebutalotofit** [comment]
Same issue here as well.

**Comment by ryokugyu** [comment]
Any solution for this? I am also facing the same issue.

**Comment by omarmaree** [comment]
same problem with me. Please any solution?

**Comment by MathCau** [comment]
Hi!
Not sure exactly what worked but here is what I did : 
- DL VC tools

- set VSINSTALLDIR = C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\Tools in my environment variable

- run on anaconda prompt the command : 
conda install -c conda-forge/label/broken keras
cd C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC
vcvarsall.bat

- open Spyder and run : from keras.models import Sequential
- got the console : 
"""
Éditeur de Spyder

Ceci est un script temporaire.
"""

from keras.models import Sequential
Using TensorFlow backend.

In [2]:

I guess it worked and the issue was the VSINSTALLDIR not being initialised in my environment variable

**Comment by amanrs** [comment]
well this was what worked for me

>conda remove keras

and then install backend engines like TensorFlow, Theano, or CNTK if you haven't

>pip install tensorflow

then install keras

>pip install keras

**Comment by naomifridman** [comment]
same issue. some things are working, but I get strange cuda errors

**Comment by amanrs** [comment]
@naomifridman did you install Cuda from vscode terminal?

**Comment by omarmaree** [comment]
I solved the problem by doing a factory reset to my pc in order to uninstall cuda, cudnn, vs2015 cause there was a conflict between versions. then, i installed python3.6 ,VS2017 with VC++2015 TOOLSET, CUDA10.0 and Cudnn7.4.1.

**Comment by rezasugiarto** [comment]
I face same problem and i think the problem because vs2015 runtime and vs 2015 win-64. If i try to uninstall it, it will remove all my env lib. I know that this is just warning, but this annoy me when activate my env

**Comment by sandysd99** [comment]
try update your conda by:-    conda update -n base -c defaults conda

**Comment by dian-ai** [comment]
this video explains a very simple way: only go for 3 commands
1- conda create --name deeplearning
2- activate deeplearning
3- conda install -c anacoda keras

https://www.youtube.com/watch?v=MIkZ6cDE53w

**Comment by educatedguessing** [comment]
I just had the same issue.
Unfortunately installing or updating VS did not help. Installing tensorflow and keras using pip did work, but conda and pip do not always go well together..
In the end, running
conda install tensorflow
conda install -c conda-forge/label/broken keras
got me a working keras env.

OS: windows 10
conda: 4.8.1
python: 3.7.6

**Comment by robisen1** [comment]
> This even happens when you try and just install theano without keras.

I have the same issue but i did not install theano. rather i already installed tensorflow and got the same issue when i installed keras

**Comment by OrbitPodium** [comment]
Close Anaconda Navidator and kill python running process.
run conda install -c anaconda keras from Anaconda prompt as administrator and the installation goes success

**Comment by Ke-V25** [comment]
> Close Anaconda Navidator and kill python running process.
> run conda install -c anaconda keras from Anaconda prompt as administrator and the installation goes success
Thanks
That worked for me

**Comment by AKhileshPothuri** [comment]
simple **pip install keras** worked for me

**Comment by rkumar399** [comment]
I am able to solve this issue after installing VS 2015 and the C# Tools from VS2015 are needed for vcvarsall.bat which is not part of default installation. Below url can be used to download VS 2015 community edition.

https://stackoverflow.com/questions/44290672/how-to-download-visual-studio-community-edition-2015-not-2017
https://go.microsoft.com/fwlink/?LinkId=532606&clcid=0x409
https://go.microsoft.com/fwlink/?LinkId=615448&clcid=0x409

**Comment by rkumar399** [comment]
I am able to solve this issue after installing VS 2015 and also C# Tools which is part of VS2015 is needed for vcvarsall.bat, C# is not part of default installation. Below urls can be used to download VS 2015 community edition.

https://stackoverflow.com/questions/44290672/how-to-download-visual-studio-community-edition-2015-not-2017
https://go.microsoft.com/fwlink/?LinkId=532606&clcid=0x409
https://go.microsoft.com/fwlink/?LinkId=615448&clcid=0x409

**Comment by mlduarte** [comment]
as per comment from @dian-ai, using `conda install -c anacoda keras` instead of `conda install -c conda-forge keras` fixed the issue for me

**Comment by devansh-srivastav** [comment]
> Close Anaconda Navidator and kill python running process.
> run conda install -c anaconda keras from Anaconda prompt as administrator and the installation goes success

Yeah It worked. Thank You

**Comment by barthelemymp** [comment]
https://stackoverflow.com/questions/57082951/anaconda-installation-error-the-system-was-unable-to-find-the-specified-regist

This solution worked for me

**Comment by rponweb** [comment]
launch Anaconda navigator-->under environments tab --> select uninstalled from the dropdown--> then search for the installed Visual studio version VS and enable it.. apply and exit
then try reinstall teraflow and keras from command..
should work

**Comment by fnandocontreras** [comment]
installing keras from anaconda prompt worked for me, i was running into the problem while using powershell in windows10

**Comment by MarkLobo** [comment]
For what is worth I had the same problem installing Keras in Anaconda. I installed TensoFlow and then Keras, in PyCharm, which was using the same Anaconda Enviroment

**Comment by yuvaraj-k** [comment]
> well this was what worked for me
> 
> > conda remove keras
> 
> and then install backend engines like TensorFlow, Theano, or CNTK if you haven't
> 
> > pip install tensorflow
> 
> then install keras
> 
> > pip install keras

this worked for me. thanks for sharing
